monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -995,7 +996,7 @@ sealed abstract class Task[+A] extends Serializable with TaskDeprecated.BinCompa
   @UnsafeBecauseImpure
   final def runSyncStepOpt(implicit s: Scheduler, opts: Options): Either[Task[A], A] = {
     val opts2 = opts.withSchedulerFeatures
-    Local.bindCurrentIf(opts2.localContextPropagation) {
+    Local.bindCurrentIf(opts2.localContextPropagation && opts2.localContextIsolateOnRun) {
```

Comment:
@Avasil So I think that this is causing the issue where concurrent read/writes of `Local` stopped working if you have `disableLocalContextIsolateOnRun`